src/js/actions/tools.js

```diff
@@ -419,9 +431,124 @@ define(function (require, exports) {
      * @private
      * @type {function()}
      */
-    var _toolModalStateChangedHandler;
+    var _toolModalStateChangedHandler,
+        _vectorMaskHandler;
     
     /**
+     * Change the tool's vector mode
+     * 
+     * @param {boolean} vectorState
+     * @return {Promise}
+     */
+    var changeVectorState = function (vectorState) {
```

Comment:
This is an incredibly small amount of JSDoc for such a long function. Please explain what this function is doing, and see if some of this can be refactored. I've noticed `[217, 217, 217]` in three different places, there has to be a way these can be organized better and make everything cleaner / less error prone.